### PR TITLE
Update Axum integration to Axum 0.5.1

### PR DESCRIPTION
**Pull request by alisenai**
## Description
As Axum is now on version `0.5.1`, I've gone ahead and updated the integration for it as my project needed it.

## Changes
Looks like `RequestParts::headers()` has changed in return value and no longer returns an `Option`. I've gone ahead and updated the locations that use this function.

This function as of `0.4`: https://docs.rs/axum/0.4.0/axum/extract/struct.RequestParts.html#method.headers

This function as of `0.5.1`: https://docs.rs/axum/0.5.1/axum/extract/struct.RequestParts.html#method.headers

Let me know if I missed something 👍🏼

### CODE REVIEW

File: integrations/axum/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ categories = ["network-programming", "asynchronous"]
 async-graphql = { path = "../..", version = "3.0.37" }
 
 async-trait = "0.1.51"
-axum = { version = "0.4", features = ["ws", "headers"] }
+axum = { version = "0.5.1", features = ["ws", "headers"] }
```
Comment (danielalvsaaker) [inline integrations/axum/Cargo.toml:18]: Why 0.5.1 instead of 0.5?

Comment (alisenai) [reply integrations/axum/Cargo.toml:18]: It's the latest version, but no other reason other than my project needing  a version >= `0.5`.
If there's a reason to use `0.5` over `0.5.1`, this pull request's fixes still apply - assuming a `0.5` in the `Cargo.toml`.

Though, the release notes lead me to think they're almost the same: https://github.com/tokio-rs/axum/releases/tag/axum-v0.5.1

Comment (danielalvsaaker) [reply integrations/axum/Cargo.toml:18]: Sorry, I misunderstood how Cargo interprets version requirements. This seems fine

Comment (alisenai) [reply integrations/axum/Cargo.toml:18]: No worries 👍🏼

**Comment by alisenai** [comment]
Looks like CI/CD is failing due to:
```
note: perhaps two different versions of crate `axum_core` are being used?
```

It is referencing `0.5.1`:
```
/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.5.1/src/handler/mod.rs:292:1
```
and  `0.4.8`:
```
/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.4.8/src/routing/method_routing.rs:586:5
```

**Comment by sunli829** [comment]
Please help update examples repo.  @alisenai  😁

**Comment by alisenai** [comment]
Sorry for the delay.
I've created a pull request updating the Axum examples to Axum version `0.5.1` here: https://github.com/async-graphql/examples/pull/51

**Comment by alisenai** [comment]
Looks like the `examples` submodule was not updated by CI (which makes sense). I've gone ahead and moved it to the latest revision which includes the fixes in https://github.com/async-graphql/examples/pull/51 👍🏼

**Comment by alisenai** [comment]
Not sure why clippy suddenly decided unrelated code is incorrect (?), but I've gone ahead and fixed that as well.

**Comment by sunli829** [comment]
Thanks! 😁

**Comment by meh** [comment]
Any chance we could get a new release with this upgrade? :panda_face:

**Comment by sunli829** [comment]
Released in `v3.0.38`😄

**Pull request merged**